### ISSUE TITLE
test_udp.ReactorShutdownInteraction needs to clean up the UDP port

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2155]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| PenguinOfDoom reported|
|-|-|
|Trac ID|trac#2155|
|Type|defect|
|Created|2006-10-06 22:01:03Z|

Patch:
```
Index: twisted/test/test_udp.py
===================================================================
--- twisted/test/test_udp.py    (revision 18400)
+++ twisted/test/test_udp.py    (working copy)
@@ -578,6 +578,7 @@
             # doesn't hang, this test is satisfied.  (There may be room for
             # another, stricter test.)
             log.flushErrors()
+            return p.stopListening()
         finished.addCallback(flushErrors)
         server.transport.write('\0' * 64, ('127.0.0.1',
                                            server.transport.getHost().port))
```

Attachments:

* [2155.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/9d9/9d985188444175d30c709f634f65967f67f1ae22/b4a02fbe4cb438fc55a7397d5b2d886d7c86e476.diff) (2020 bytes) - added by PenguinOfDoom on 2006-10-08 23:32:25Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2155 2155
type__defect defect
reporter__PenguinOfDoom PenguinOfDoom
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1160172063000000 1160172063000000
changetime__1160351269000000 1160351269000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to PenguinOfDoom|
|-|-|

This doesn't look correct to me.  The port should be created in a setUp method and shut down in a tearDown method.

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|PenguinOfDoom set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to PenguinOfDoom|
|-|-|

Looks good, but please fix indentation of the transport.write line near the end of the test method before committing.

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|pahan set status to `closed`|
|-|-|

(In [18411]) Add port setup/teardown to ReactorShutdownInteraction

Author: pahan
Reviewer: exarkun
Fixes [#2155](https://github.com/twisted/twisted/issues/2155)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
